### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by tamana-Balkhi**
# Project requirements

- Create models
- Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- Set up associations between models.
- Use models to insert data
- Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- Open the rails console for your project.

Create at least one user by running the following code:
```
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
```
```
second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
```
Create at least 4 posts written by one of the users you created by running the following code:
```
first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')
```
Create at least 6 posts comments for one of the posts you created by running the following code:
```
Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )
```
Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.
Create custom methods
Your models should include some additional methods.
## Users
A method that returns the 3 most recent posts for a given user.

## Posts
A method that updates the posts counter for a user.
A method which returns the 5 most recent comments for a given post.

## Comments
A method that updates the comments counter for a post.
## Likes
A method that updates the likes counter for a post.
Go to rails c and check if your methods are working.

## Create models.
Use models to insert data.
Create custom methods.
Check if methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+  validates :likes_counter, comparison: { greater_than_or_equal_to: 0 }, numericality: { only_integer: true }
+
+  def last_five_comments
+    comments.order('created_at Desc').limit(5)
+  end
+
+  private
+
+  def update_posts_counter
+    author.update(postCounter: author.posts.length)
```
Comment (DammyShittu) [inline app/models/post.rb:15]: - **[OPTIONAL]** I think we can make this shorter by using the `increment!` method.

  ```
  def update_posts_counter
    author.increment!(:postCounter)
  end
  ```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
...
-  belongs_to :post, class_name: 'Post', foreign_key: :post_id
+  after_save :update_likes_counter
+
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  private
+
+  def update_likes_counter
+    post.update(likes_counter: post.likes.length)
```
Comment (DammyShittu) [inline app/models/like.rb:10]: - **[OPTIONAL]** Following the feedback earlier I think we can modify this as well.

  ```
  def update_likes_counter
    post.increment!(:likes_counter)
  end
  ```

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
+  after_save :update_comments_counter
+
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  private
+
+  def update_comments_counter
+    post.update(comments_counter: post.comments.length)
+  end
```
Comment (DammyShittu) [inline app/models/comment.rb:11]: - **[OPTIONAL]** We can modify this method also to be shorter.

**Review by DammyShittu — APPROVED** [review]
Hi @tamana-Balkhi,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (tamana-Balkhi) [reply app/models/post.rb:15]: thanks, @DammyShittu  for your great suggestions I will apply them right now

**Pull request merged**